test/distributed/org/apache/cassandra/distributed/test/SSTableIdGenerationTest.java

```diff
@@ -0,0 +1,229 @@
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, v int, primary key (pk))");
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+            assertSSTablesCount(cluster, "tbl", 2, 0);
+            verfiySSTableActivity(cluster, true);
+
+            waitOn(cluster.get(1).shutdown());
+            cluster.get(1).config().set(ENABLE_UUID_FIELD_NAME, true);
+            cluster.get(1).startup();
+
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+
+            assertSSTablesCount(cluster, "tbl", 2, 3);
+
+            assertThat(cluster.get(1).executeInternalWithResult(String.format("SELECT * FROM %s.%s", KEYSPACE, "tbl")).toObjectArrays()).hasSize(4);
+        }
+    }
+
+    @Test
+    public void testRestartWithUUIDDisabled() throws IOException
+    {
+        try (Cluster cluster = init(Cluster.build(1).withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, true)).start()))
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, v int, primary key (pk))");
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+            assertSSTablesCount(cluster, "tbl", 0, 2);
+            verfiySSTableActivity(cluster, false);
+
+            waitOn(cluster.get(1).shutdown(true));
+            cluster.get(1).config().set(ENABLE_UUID_FIELD_NAME, false);
+
+            Assertions.assertThatExceptionOfType(RuntimeException.class)
+                      .isThrownBy(() -> cluster.get(1).startup())
+                      .withCauseInstanceOf(SystemExitException.class);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCompactionStrategiesWithMixedSSTables() throws Exception
+    {
+        testCompactionStrategiesWithMixedSSTables(SizeTieredCompactionStrategy.class, DateTieredCompactionStrategy.class, TimeWindowCompactionStrategy.class, LeveledCompactionStrategy.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void testCompactionStrategiesWithMixedSSTables(Class<? extends AbstractCompactionStrategy>... compactionStrategyClasses) throws Exception
+    {
+        try (Cluster cluster = init(Cluster.build(1).withDataDirCount(1).withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, false)).start()))
+        {
+            for (Class<? extends AbstractCompactionStrategy> compactionStrategyClass : compactionStrategyClasses)
+            {
+                String tableName = "tbl_" + compactionStrategyClass.getSimpleName().toLowerCase();
+
+                cluster.schemaChange(String.format("CREATE TABLE %s.%s (pk int, v int, primary key (pk)) WITH compaction = {'class':'%s', 'enabled': 'false'}", KEYSPACE, tableName, compactionStrategyClass.getCanonicalName()));
```

Comment:
Nit: can we break this long line? Maybe something like:
```suggestion
                cluster.schemaChange(String.format("CREATE TABLE %s.%s (pk int, v int, primary key (pk)) " +
                                                   "WITH compaction = {'class':'%s', 'enabled': 'false'}",
                                                   KEYSPACE, tableName, compactionStrategyClass.getCanonicalName()));
```